packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,602 @@
+
+In the config they can also tell you a variety of things about them, such as the name of their application or a url for an icon of their application. You can use these pieces of information to customize your wallet's user experience should you desire to do so.
+
+Your wallet having a visual distinction from the application, but still a seamless and connected experience is our goal here.
+
+Whether your authentication process happens using a webpage with the `IFRAME/RPC`, `POP/RPC` or `TAB/RPC` methods, or using a backchannel to an API with the `HTTP/POST` method, the handshake is the same. The same messages are sent in both methods, however the transport mechanism changes. For `IFRAME/RPC`, `POP/RPC` or `TAB/RPC` methods, the transport is `window.postMessage()`, with the `HTTP/POST` method, the tranport is HTTP post messages. 
+
+As always, you must never trust anything you receive from an application. Always do your due-dilligence and be alert as you are the users first line of defense against potentially malicious applications.
+
+### Authenticate your User 
+
+It's important that you are confident that the user is who the user claims to be.
+
+Have them provide enough proof to you that you are okay with passing their details back to FCL.
+Using Blocto as an example, an authentication code is sent to the email a user enters at login.
+This code can be used as validation and is everything Blocto needs to be confident in the users identity.
+
+### Once you know who your User is
+
+Once you're confident in the users identity, we can complete the authentication process.
+
+The authentication process is complete once FCL receives back a response that configures FCL with FCL Services for the current user. This response is extremeley important to FCL. At its core it tells FCL who the user is, and then via the included services it tells FCL how the user authenticated, how to request transaction signatures, how to get a personal message signed and the user's email and other details if requested. In the future it may also inlude many more things!
+
+You can kind of think of FCL as a plugin system. But since those plugins exist elsewhere outside of FCL, FCL needs to be configured with information on how to communicate with those plugins.
+
+What you are sending back to FCL is everything that it needs to communicate with the plugins that you are supplying.
+Your wallet is like a plugin to FCL, and these details tell FCL how to use you as a plugin.
+
+Here is an example of an authentication resonse:
+
+```javascript
+// IN WALLET AUTHENTICATION FRAME
+import {WalletUtils} from "@onflow/fcl"
+
+WalletUtils.sendMsgToFCL("FCL:VIEW:RESPONSE", {
```

Comment:
Maybe change this to `WalletUtils.approve`?

Reply:
Updated!